larpandoracontent/LArThreeDReco/LArCosmicRay/UnambiguousDeltaRayTool.cc

```diff
@@ -0,0 +1,114 @@
+/**
+ *  @file   larpandoracontent/LArThreeDReco/LArCosmicRay/UnambiguousDeltaRayTool.cc
+ *
+ *  @brief  Implementation of the unambiguous delta ray tool class.
+ *
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+
+#include "larpandoracontent/LArHelpers/LArClusterHelper.h"
+#include "larpandoracontent/LArHelpers/LArPfoHelper.h"
+
+#include "larpandoracontent/LArThreeDReco/LArCosmicRay/UnambiguousDeltaRayTool.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+UnambiguousDeltaRayTool::UnambiguousDeltaRayTool() :
+    m_maxSeparation(2.f),
+    m_minNConnectedClusters(1)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool UnambiguousDeltaRayTool::Run(ThreeViewDeltaRayMatchingAlgorithm *const pAlgorithm, TensorType &overlapTensor)
+{
+    if (PandoraContentApi::GetSettings(*pAlgorithm)->ShouldDisplayAlgorithmInfo())
+       std::cout << "----> Running Algorithm Tool: " << this->GetInstanceName() << ", " << this->GetType() << std::endl;
+
+    bool changesMade(false);
+
+    TensorType::ElementList elementList;
+    overlapTensor.GetUnambiguousElements(true, elementList);
+
+    this->ExamineUnambiguousElements(pAlgorithm, elementList, changesMade);
+
+    return changesMade;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void UnambiguousDeltaRayTool::ExamineUnambiguousElements(ThreeViewDeltaRayMatchingAlgorithm *const pAlgorithm, TensorType::ElementList &elementList, bool &changesMade)
```

Comment:
I think you can just return a boolean here, rather than passing in the reference variable - the logic of the function doesn't appear to be dependent on the value of the boolean that gets passed in